### PR TITLE
L4 Telemetry is backwards

### LINKED ISSUE #5822: L4 Telemetry is "backwards"
Arguably, istio metrics are backwards (https://github.com/istio/istio/pull/32549), but :shrug: 

What I did:

Shell ---> Echo. 
I sent a few bytes from shell to echo and got 100s back

In the graph, on the edge from shell --> Echo I see: Sent: 14. Recv: 0.1

I expect: Sent: 0.1, Recv: 14

### PR DESCRIPTION
**Pull request by josunect**
Fixes #5822

### CODE REVIEW

**Comment by jshaughn** [comment]
I took a quick look and nice job, I think you found the two places I was talking about when I said, "... in the graph itself we may be able to make a very local change. On the server side, we may also be able to make a local change".  I'll follow up with you on Monday...

**Comment by josunect** [comment]
> I took a quick look and nice job, I think you found the two places I was talking about when I said, "... in the graph itself we may be able to make a very local change. On the server side, we may also be able to make a local change". I'll follow up with you on Monday...

Sure, thanks @jshaughn . 
I will be reviewing the tests failures.

**Comment by josunect** [comment]
Example with mTLS enabled: 

image

File: graph/api/api_test.go [hunk 1]
```diff
@@ -2938,7 +2938,10 @@ func TestRatesNodeGraphTotal(t *testing.T) {
...
 	v6 := model.Vector{}
 
-	q7 := `round(sum(rate(istio_tcp_sent_bytes_total{reporter="source",source_workload_namespace="bookinfo",source_workload="productpage-v1"} [600s])) by (source_cluster,source_workload_namespace,source_workload,source_canonical_service,source_canonical_revision,destination_cluster,destination_service_namespace,destination_service,destination_service_name,destination_workload_namespace,destination_workload,destination_canonical_service,destination_canonical_revision,response_flags) > 0,0.001)`
+	q6a := `round(sum(rate(istio_tcp_received_bytes_total{reporter="destination",destination_workload_namespace="bookinfo",destination_workload="productpage-v1"} [600s])) by (source_cluster,source_workload_namespace,source_workload,source_canonical_service,source_canonical_revision,destination_cluster,destination_service_namespace,destination_service,destination_service_name,destination_workload_namespace,destination_workload,destination_canonical_service,destination_canonical_revision,response_flags) > 0,0.001)`
```
Comment (jshaughn) [inline graph/api/api_test.go:2941]: Why did we need to add an extra query?

File: business/metrics_definitions.go [hunk 2]
```diff
@@ -56,12 +56,12 @@ var istioMetrics = []istioMetric{
 	},
 	{
 		kialiName: "tcp_received",
-		istioName: "istio_tcp_received_bytes_total",
+		istioName: "istio_tcp_sent_bytes_total", // L4 telemetry is backwards
```
Comment (jshaughn) [inline business/metrics_definitions.go:59]:
```suggestion
		istioName: "istio_tcp_sent_bytes_total", // L4 telemetry is backwards, see https://github.com/istio/istio/issues/32399
```

File: business/metrics_definitions.go [hunk 3]
```diff
@@ -56,12 +56,12 @@ var istioMetrics = []istioMetric{
...
 		isHisto:   false,
 	},
 	{
 		kialiName: "tcp_sent",
-		istioName: "istio_tcp_sent_bytes_total",
+		istioName: "istio_tcp_received_bytes_total", // L4 telemetry is backwards
```
Comment (jshaughn) [inline business/metrics_definitions.go:64]:
```suggestion
		istioName: "istio_tcp_received_bytes_total", // L4 telemetry is backwards, see https://github.com/istio/istio/issues/32399
```

File: graph/telemetry/istio/istio.go [hunk 4]
```diff
@@ -742,11 +742,11 @@ func buildNodeTrafficMap(cluster, namespace string, n *graph.Node, o graph.Telem
 
 		switch o.Rates.Tcp {
 		case graph.RateReceived:
-			metrics = []string{"istio_tcp_received_bytes_total"}
```
Comment (jshaughn) [inline graph/telemetry/istio/istio.go:745]: Above the switch stmt can we add:

// L4 telemetry is backwards, see https://github.com/istio/istio/issues/32399

**Review by jshaughn — CHANGES_REQUESTED** [review]
It looks strange to see the numbers and metrics reversed, but I think it is correct.  I had one question and am requesting just a few changes in comments.

**Review by jshaughn — APPROVED** [review]
Looks good, nice job!  Assuming the CI passes we can merge.

**Pull request merged**